tests/sles4sap/hana.pm

```diff
@@ -0,0 +1,139 @@
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+# Summary: Checks HANA Wizard installation on Gnome
+# Requires: ENV variable MEDIA pointing to installation media, QEMURAM=32768
+# Maintainer: Ricardo Branco <rbranco@suse.de>
+
+use base 'sles4sap';
+use strict;
+use testapi;
+use utils 'turn_off_gnome_screensaver';
+use utils 'type_string_slow';
+
+sub run {
+    my ($self) = @_;
+    my ($proto, $path) = split m|://|, get_required_var('MEDIA');
+
+    die "Currently supported protocols are nfs and smb"
+      unless ($proto eq 'nfs' or $proto eq 'smb');
+
+    my $QEMURAM = get_required_var('QEMURAM');
+    die "QEMURAM=$QEMURAM. QEMURAM must be at least 32768" if $QEMURAM < 32768;
+
+    # Add host's IP to /etc/hosts
+    select_console 'root-console';
+    assert_script_run 'echo $(ip -4 addr show dev eth0 | sed -rne "/inet/s/[[:blank:]]*inet ([0-9\.]*).*/\1/p") $(hostname) >> /etc/hosts';
+    select_console 'x11', await_console => 0;
+
+    x11_start_program('xterm -geometry 160x45+5+5', target_match => 'xterm-susetest');
+    turn_off_gnome_screensaver;
+    type_string "killall xterm\n";
+
+    assert_screen 'generic-desktop';
+    x11_start_program 'yast2 sap-installation-wizard';
+    assert_screen 'sap-installation-wizard';
+
+    # Choose nfs://
+    send_key 'tab';
+    send_key_until_needlematch 'sap-wizard-proto-' . $proto . '-selected', 'down';
+    send_key 'alt-p';
+    type_string_slow "$path", wait_still_screen => 1;
+    # Next
+    send_key 'tab';
+    send_key 'alt-n';
+
+    assert_screen 'sap-wizard-copying-media';
+
+    # "Do you use a Supplement/3rd-Party SAP software medium?"
+    assert_screen 'sap-wizard-supplement-medium', 3000;
+    # No
+    send_key 'alt-n';
+    assert_screen 'sap-wizard-additional-repos';
+    # Next
+    send_key 'alt-n';
+
+    # Don't change this. The needle has this SID.
+    my $sid      = 'NDB';
+    my $password = 'Qwerty_123';
+    set_var('PASSWORD', $password);
+
+    assert_screen 'sap-wizard-hana-system-parameters';
+    # SAP SID
+    send_key 'alt-s';
+    type_string $sid;
+    # SAP Master Password
+    send_key 'alt-a';
+    type_password $password;
+    send_key 'tab';
+    type_password $password;
+    # Ok
+    send_key 'alt-o';
+
+    set_var('SAPADM', lc($sid) . 'adm');
+
+    assert_screen 'sap-wizard-performing-installation', 60;
+
+    # "Are there more SAP products to be prepared for installation?"
+    assert_screen 'sap-wizard-profile-ready', 300;
+    # No
+    send_key 'alt-n';
+
+    # "Do you want to continue the installation?"
+    # "Your system does not meet the requirements..."
+    assert_screen 'sap-wizard-continue-installation';
+    # Yes
+    send_key 'alt-y';
+
+    assert_screen 'sap-product-installation';
+
+    assert_screen [qw(sap-wizard-installation-summary sap-wizard-finished sap-wizard-failed sap-wizard-error)], 4000;
```

Comment:
Same question as in MR: the  sap-wizard-error error seems more a YaST bug, should be trapped more clearly by YaST I think. Did you already opened a bug for this? If so, I think it's a good idea to trap the bug number.

Reply:
That particular error was "No space left on device".  It's not a YaST bug. This needle is useful to catch this and other bugs that may interfere with YaST.

Reply:
I understand, but as YaST is failing (so not trap this error correctly) I think it's good to open a bug for this :) Even if it's not a common case.
And the softfailed code could be added after.

@alvarocarvajald, what do you think about this?

Reply:
Replying on the MR